spacy/ml/tb_framework.pyx

```diff
@@ -133,27 +145,36 @@ def init(
 
 
 def forward(model, docs_moves: Tuple[List[Doc], TransitionSystem], is_train: bool):
-    nF = model.get_dim("nF")
-    tok2vec = model.get_ref("tok2vec")
+    beam_width = model.attrs["beam_width"]
     lower_pad = model.get_param("lower_pad")
-    lower_W = model.get_param("lower_W")
+    tok2vec = model.get_ref("tok2vec")
+
+    docs, moves = docs_moves
+    states = moves.init_batch(docs)
+    tokvecs, backprop_tok2vec = tok2vec(docs, is_train)
+    tokvecs = model.ops.xp.vstack((tokvecs, lower_pad))
+    feats, backprop_feats = _forward_precomputable_affine(model, tokvecs, is_train)
+    seen_mask = _get_seen_mask(model)
```

Comment:
I am wondering if we can correctly cast this, since `np.ndarray` is a C type. I think the best solution is probably to have `_forward_greedy_cpu` take `Floats1d` and then internally convert it to the needed type.

Reply:
Although that would probably still give an IDE error 🤔 .